File: 2024-2/0684/unidad06.c_Web_03/SofPub/SoftPubPersonaDA/DAO/EspecialidadDAO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaModel;

namespace SoftPubPersonaDA.DAO
{
    public interface EspecialidadDAO
    {
        int insertar(Especialidad especialidad);

        int eliminar(Especialidad especialidad);

        int modificar(Especialidad especialidad);

        BindingList<object> listarTodos();
    }
}

File: 2024-2/0682/unidad07.e - Programación distribuida - RMI - con BD/SoftPub_Back-end/ServidorRMI/SoftPubPersonaDA/src/pe/edu/pucp/softpub/persona/dao/EspecialidadDAO.java
package pe.edu.pucp.softpub.persona.dao;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.model.Especialidad;

/**
 *
 * @author andres
 */
public interface EspecialidadDAO {

    public Integer insertar(Especialidad especialidad);

    public Integer modificar(Especialidad especialidad);

    public Integer eliminar(Especialidad especialidad);

    public ArrayList<Especialidad> listarTodos();

    public Especialidad obtenerPorId(Integer especialidad);
}
